### PR TITLE
js: Force namespaces to be lowercase

### PR DESCRIPTION
**Pull request by HongjianZhang**
GCE doesn't accept capitalized namespaces.
In vet(), throw an error if provided namespace has capitalized letter.

### CODE REVIEW

File: docs/_GettingStarted.md [hunk 1]
```diff
@@ -107,7 +107,7 @@ $ git clone https://github.com/quilt/nginx.git
 ```
 
 The blueprint is the `main.js` file in the nginx directory; take a look at this
-file if you'd like an to see an example of what blueprints look like.  This
+file if you'd like to see an example of what blueprints look like.  This
```
Comment (kklin) [inline docs/_GettingStarted.md]: This should be in another commit since it's not really related to the namespace capitalization issue

File: stitch/bindings.js [hunk 2]
```diff
@@ -220,14 +220,18 @@ Deployment.prototype.toQuiltRepresentation = function toQuiltRepresentation() {
 };
 
 /**
- * Checks if all referenced containers in connections and load balancers are
- * really deployed.
+ * Checks if namespace is lower case, and if all referenced containers
```
Comment (kklin) [inline stitch/bindings.js]: I think this would read a little better as "if _the_ namespace"

File: docs/_GettingStarted.md [hunk 3]
```diff
@@ -107,7 +107,7 @@ $ git clone https://github.com/quilt/nginx.git
 ```
```
Comment (kklin) [inline docs/_GettingStarted.md]: A couple comments on the commit message:
- We've sort of standardized on categorizing this part of the code as `js` rather than `bindings`. It doesn't really matter, but it makes `git log` a little easier to follow if it's consistent.
- The first letter after the colon in the commit title should be capitalized.
- The body should be fleshed out more, and be written in prose.

So, I'd write the commit message to be:
```
js: Force namespaces to be lowercase

GCE doesn't accept capitalized namespaces. Before this commit, if a user tried
to use a namespace with a capital letter on GCE, the daemon would log an error while
booting machines. This could be hard to debug, so this commit makes it so `quilt run`
fails when the blueprint specifies a namespace with any capitalized letters.
```

File: stitch/bindings.js [hunk 4]
```diff
@@ -220,14 +220,18 @@ Deployment.prototype.toQuiltRepresentation = function toQuiltRepresentation() {
...
+ * in connections and load balancers are really deployed.
  * @private
  *
  * @param {Deployment} deployment - A deployment object.
  * @returns {void}
  */
 function vet(deployment) {
+  if (deployment.namespace !== deployment.namespace.toLowerCase()) {
+    throw new Error(`namespace "${deployment.namespace}" ` +
```
Comment (kklin) [inline stitch/bindings.js]: I think this error message could be more user-friendly. Something like `namespace "foo" contains uppercase letters. Namespaces must be lowercase.`. (@kayousterhout may have opinions on this)

File: stitch/bindings_test.js [hunk 5]
```diff
@@ -717,14 +717,19 @@ describe('Bindings', () => {
...
+      deployment = createDeployment({ namespace: 'mynamespace' });
       expect(deployment.toQuiltRepresentation().namespace).to.equal(
-        'myNamespace');
+        'mynamespace');
     });
     it('default namespace', () => {
       expect(deployment.toQuiltRepresentation().namespace).to.equal(
         'default-namespace');
     });
+    it('should error when given namespace contains upper case letters', () => {
```
Comment (kklin) [inline stitch/bindings_test.js]: This test would be a little more appropriate in the `vet` section IMO

**Comment by kklin** [comment]
Oh, looks like you need to rebase as well because @kayousterhout merged some renaming stuff this weekend.

Also, I think the author in your commit should be tweaked a bit. I see
```
HongjianZhang <zhanghongjian.1995@gmail.com>
```
I don't remember if we decided whether to use your Quilt email for git (@kayousterhout?), but your name should have a space between first and last name.

**Comment by kayousterhout** [comment]
Re: email, I don't have strong feelings about which address to use (either seems fine to me)

**Comment by kklin** [comment]
Either is fine with me as well

File: stitch/bindings.js (hunk 4 above)
Comment (kayousterhout) [reply stitch/bindings.js]: Your suggestion looks great @kklin

File: blueprint/bindings_test.js [hunk 6]
```diff
@@ -658,6 +658,12 @@ describe('Bindings', () => {
       foo = new LoadBalancer('foo', []);
       deployment.deploy([foo]);
     });
+    it('should error when given namespace contains upper case letters', () => {
+      deployment = createDeployment({ namespace: 'BadNamespace' });
+      expect(() => deployment.toQuiltRepresentation())
```
Comment (kklin) [inline blueprint/bindings_test.js]: This could just be `expect(deploy)`

Comment (HongjianZhang) [reply blueprint/bindings_test.js]: Made the change, and amend to the commit.
